obs-packaging/build_from_docker.sh

```diff
@@ -4,50 +4,74 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 #
+[ -z "${DEBUG}" ] || set -o xtrace
 
-set -x
-set -e
+set -o errexit
+set -o nounset
+set -o pipefail
 
 script_dir=$( cd $( dirname "${BASH_SOURCE[0]}" ) && pwd )
+script_name="$(basename "${BASH_SOURCE[0]}")"
 cache_dir=${PWD}/obs-cache
 #where packaing repo lives
 packaging_repo_dir=$(cd "${script_dir}/.." && pwd )
 #where results will be stored
 host_datadir="${PWD}/pkgs"
 obs_image="obs-kata"
 export USE_DOCKER=1
+http_proxy=${http_proxy:-}
+https_proxy=${https_proxy:-}
+no_proxy=${no_proxy:-}
+PUSH=${PUSH:-}
 
-if command -v go; then
-	export GO_ARCH=$(go env GOARCH)
-else
-	export GO_ARCH=amd64
-	echo "Go not installed using $GO_ARCH to install go in dockerfile"
-fi
+GO_ARCH=$(go env GOARCH)
+export GO_ARCH
 
-export GO_ARCH=$(go env GOARCH)
-sudo docker build \
-	--build-arg http_proxy="${http_proxy}" \
-	--build-arg https_proxy="${https_proxy}" \
-	-t $obs_image ${script_dir}
+docker_run(){
+	local cmd="$@"
+	sudo docker run \
+		--rm \
+		-v "${HOME}/.ssh":/root/.ssh \
+		-v "${HOME}/.gitconfig":/root/.gitconfig \
+		-v /etc/profile:/etc/profile \
+		--env http_proxy="${http_proxy}" \
+		--env https_proxy="${https_proxy}" \
+		--env no_proxy="${no_proxy}" \
+		--env PUSH="${PUSH}" \
```

Comment:
Don't you need a `--env DEBUG="${DEBUG}"` here too?

Reply:
that is true! added